src/views/Account/AccountBalances.js

```diff
@@ -1,225 +1,61 @@
-  & > p {
-    font-size: ${fonts.size.medium};
-    font-weight: ${fonts.weight.semibold};
-    font-family: ${fonts.family.SFMono};
-  }
-  & > p:first-child {
-    font-family: ${fonts.family.SFProText};
-    justify-content: flex-start;
-  }
-  @media screen and (${responsive.sm.max}) {
-    & p {
-      font-size: ${fonts.size.h6};
-    }
-  }
-`;
-
-const StyledShowMoreTokens = styled(StyledToken)`
-  grid-template-columns: 100%;
-  min-height: 0;
-  min-width: 0;
-  padding: 0;
-  position: relative;
-  cursor: pointer;
-  text-align: left;
-  justify-content: flex-start;
-  font-family: ${fonts.family.SFProText};
-  font-weight: ${fonts.weight.medium};
-  font-size: 13px;
-  color: rgb(${colors.grey});
-  margin-top: -1px;
-  padding-left: 19px;
-
-  @media (hover: hover) {
-    &:hover p {
-      opacity: 0.7;
-    }
-  }
-`;
+import {
+    StyledGrid,
+    StyledLabelsRow,
+    StyledLabels,
+    StyledEthereum,
+    StyledToken,
+    StyledAsset,
+    StyledPercentage,
+    StyledLastRow,
+    StyledShowMoreTokens,
+    StyledHideIcon,
+    StyledShowIcon
+} from './styles';
 
 class AccountBalances extends Component {
   state = {
     disableToggle: false,
     showMoreTokens: false,
+    hoveredAsset: null
   };
   onShowMoreTokens = () => {
     this.setState({ showMoreTokens: !this.state.showMoreTokens });
   };
+  onAssetHover = address => {
+    this.setState({ hoveredAsset: address });
+  }
+  onHideAsset = address => {
+    this.props.accountHideAsset(address); 
+    this.setState({ hoveredAsset: null });
+  }
+  onShowAsset = address => {
+    this.props.accountShowAsset(address);
+    this.setState({ hoveredAsset: null });
+  }
   render() {
     if (!this.props.accountInfo.assets) return null;
-    const ethereum = this.props.accountInfo.assets.filter(
-      asset => asset.symbol === 'ETH',
-    )[0];
-    const tokens = this.props.accountInfo.assets.filter(
-      asset => asset.symbol !== 'ETH' && typeof asset === 'object' && !!asset,
-    );
-    if (tokens.length && tokens.length < 5 && !this.state.disableToggle) {
-      this.setState({ disableToggle: true });
-    }
-    const tokensWithHighMarketValue = tokens.filter(
-      asset => asset.symbol !== 'ETH' && hasHighMarketValue(asset),
-    );
-    const tokensWithLowMarketValue = tokens.filter(
-      asset => asset.symbol !== 'ETH' && hasLowMarketValue(asset),
-    );
-    const tokensWithNoMarketValue = tokens.filter(asset => !asset.native);
-    let tokensAlwaysDisplay = tokensWithHighMarketValue;
-    let tokensToggleDisplay = [
-      ...tokensWithLowMarketValue,
-      ...tokensWithNoMarketValue,
-    ];
-    if (this.state.disableToggle) {
-      tokensAlwaysDisplay = [...tokensAlwaysDisplay, ...tokensToggleDisplay];
-      tokensToggleDisplay = [];
-    }
-    const allLowMarketTokensHaveNoValue =
-      tokensWithNoMarketValue.length === tokensToggleDisplay.length;
+
+    const emptyBalanceDisplay = '———';
```

Comment:
Will probably create a `consts` file on an upper directory. This constant is used in other places as well.